processing/src/main/java/io/druid/query/aggregation/first/StringFirstAggregateCombiner.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.first;
+
+import io.druid.query.aggregation.ObjectAggregateCombiner;
+import io.druid.segment.ColumnValueSelector;
+
+import javax.annotation.Nullable;
+
+public class StringFirstAggregateCombiner extends ObjectAggregateCombiner<String>
+{
+  private String firstString;
+
+  @Override
+  public void reset(ColumnValueSelector selector)
+  {
+    firstString = (String) selector.getObject();
+  }
+
+  @Override
+  public void fold(ColumnValueSelector selector)
+  {
+    // Nothing to do. It needs to keep the first string value.
```

Comment:
We don't have to change the contract here. It may cause side effects so we should check before changing any contracts. 

What I suggest here is, since we can't guarantee that `reset()` is always called before `fold()`, we should check `reset()` is called or not and call it (or initialize some initial state) in `fold()` if needed.

Reply:
ok!